src/ConnectionPool.jl

```diff
@@ -348,18 +357,21 @@ function IOExtras.closeread(t::Transaction)
     @require isreadable(t)
 
     @v1_3 lock(t.c.readlock)
-    t.readbusy[] = false
-    Threads.atomic_add!(t.c.readcount, 1)         ;@debug 2 "✉️  Read done:  $t"
-    t.c.readbusy[] = false
-    notify(t.c.readlock)
-    @v1_3 unlock(t.c.readlock)
+    try
+        t.readbusy = false
+        t.c.readcount += 1         ;@debug 2 "✉️  Read done:  $t"
+        t.c.readbusy = false
+        notify(t.c.readlock)
+        @ensure !isreadable(t)
+    finally
+        @v1_3 unlock(t.c.readlock)
+    end
     release(t.c)
 
     if !isbusy(t.c)
```

Comment:
This is the race-y statement, the `ensure` was probably OK (as you told me elsewhere, it's a one-shot Event)